docs/fundamentals/aggregation-builder.txt

```diff
@@ -0,0 +1,513 @@
+          "name": "Ellis Lee",
+          "occupation": "designer",
+          "birthday": {
+            "$date": {
+              "$numberLong": "834019200000"
+            }
+          }
+        },
+        {
+          "_id": ...,
+          "name": "Eliud Nkosana",
+          "occupation": "engineer",
+          "birthday": {
+            "$date": {
+              "$numberLong": "449884800000"
+            }
+          }
+        },
+        {
+          "_id": ...,
+          "name": "Bran Steafan",
+          "occupation": "engineer",
+          "birthday": {
+            "$date": {
+              "$numberLong": "894326400000"
+            }
+          }
+        },
+        {
+          "_id": ...,
+          "name": "Alda Gröndal",
+          "occupation": "engineer",
+          "birthday": {
+            "$date": {
+              "$numberLong": "1009843200000"
+            }
+          }
+        }
+      ]
+
+.. _laravel-aggregation-project-stage-example:
+
+Project Stage Example
+~~~~~~~~~~~~~~~~~~~~~
+
+You can chain the ``project()`` method to your aggregation pipeline to specify
+which fields from the documents to display by this stage.
+
+To specify fields to include, pass the name of a field and a truthy value,
+such as ``1`` or ``true``. All other fields are omitted from the output.
+
+Alternatively, to specify fields to exclude, pass each field name and
+a falsy value, such as ``0`` or ``false``. All other fields are included in
+the output.
+
+.. tip::
+
+   When you specify fields to include, the ``_id`` field is included by default.
+   To exclude the ``_id`` field, explicitly exclude it in the projection stage.
+
+This example shows how to use the ``project()`` method aggregation stage to
+include only the ``name`` field and exclude all other fields from the output.
+Click the :guilabel:`{+code-output-label+}` button to see the data returned by
+running the code:
+
+.. io-code-block::
+
+   .. input:: /includes/fundamentals/aggregation/AggregationsBuilderTest.php
+      :language: php
+      :dedent:
+      :start-after: begin aggregation project stage
+      :end-before: end aggregation project stage
+
+   .. output::
+      :language: json
+      :visible: false
+
+      [
+        { "name": "Alda Gröndal" },
+        { "name": "Francois Soma" },
+        { "name": "Janet Doe" },
+        { "name": "Eliud Nkosana" },
+        { "name": "Bran Steafan" },
+        { "name": "Ellis Lee" }
+      ]
+
+
+.. _laravel-aggregation-pipeline-example:
+
+Aggregation Pipeline Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This aggregation pipeline example chains multiple stages. Each stage runs
+on the output retrieved from each preceding stage. In this example, the
+stages perform the following operations sequentially:
+
+- Match all documents in the collection since the ``match()`` stage
+  is omitted.
+- Add the ``year`` field to the documents and set the value to the year
+  extracted from the ``birthday`` field.
```

Comment:
S: could be more descriptive
```suggestion
- Add the ``birth_year`` field to the documents and set the value to the year
  extracted from the ``birthday`` field.
```

Reply:
If you change this, change the `year_avg` field to `birth_year_avg`

Reply:
Thanks, will update the example to use birth_year and birth_year_avg field names.